test/SymbolGtest.cpp

```diff
@@ -0,0 +1,105 @@
+////////////////////////////////////////////////////////////////////////////
+/**
+ *  @file   SymbolGtest.cpp
+ *  @date   Wed February 23 2022
+ *  @brief  Unit test suite for the Sample class
+ */
+////////////////////////////////////////////////////////////////////////////
+
+#include <limits>
+#include <sstream>
+
+#include<gtest/gtest.h>
+#include<gmock/gmock.h>
+#include "../inc/Symbol.h"
+
+using namespace testing;
+
+class SymbolFixture: public ::testing::Test
+{
+    protected:
+        calculator::Symbol<int> intSymbolA;
+        calculator::Symbol<int> intSymbolBBound;
+        calculator::Symbol<int> intSymbolCBound;
+        calculator::Symbol<int> intSymbolDBound;
+        calculator::Symbol<int> intSymbolEBound;
+        calculator::Symbol<int> intSymbolFBound;
+        calculator::Symbol<int> intSymbolZ;
+        
+        calculator::Symbol<double> doubleSymbolA;
+        calculator::Symbol<double> doubleSymbolBBound;
+        calculator::Symbol<double> doubleSymbolCBound;
+        calculator::Symbol<double> doubleSymbolDBound;
+        calculator::Symbol<double> doubleSymbolEBound;
+        calculator::Symbol<double> doubleSymbolFBound;
+        calculator::Symbol<double> doubleSymbolZ;
+
+        SymbolFixture(): 
+                intSymbolA{'a'},
+                intSymbolBBound{'B'}, 
+                intSymbolCBound{'c'},
+                intSymbolDBound{'d'},
+                intSymbolEBound{'e'},
+                intSymbolFBound{'f'},
+                intSymbolZ{'Z'},
+                doubleSymbolA{'A'},
+                doubleSymbolBBound{'b'},
+                doubleSymbolCBound{'C'},
+                doubleSymbolDBound{'d'},
+                doubleSymbolEBound{'e'},
+                doubleSymbolFBound{'f'},
+                doubleSymbolZ{'z'}
+        {
+            intSymbolBBound.setValue(-12);
+            intSymbolCBound.setValue(0);
+            intSymbolDBound.setValue(42);
+            intSymbolEBound.setValue(std::numeric_limits<int>::max());
+            intSymbolFBound.setValue(std::numeric_limits<int>::min());
+
+            doubleSymbolBBound.setValue(-123.45);
+            doubleSymbolCBound.setValue(0.0);
+            doubleSymbolDBound.setValue(42.42e10);
+            doubleSymbolEBound.setValue(std::numeric_limits<double>::max());
+            doubleSymbolFBound.setValue(std::numeric_limits<double>::min());
+        }
+};
+TEST_F(SymbolFixture, hasValue)
+{
+    EXPECT_FALSE(intSymbolA.hasValue());
+    EXPECT_TRUE(intSymbolBBound.hasValue());
+    EXPECT_FALSE(doubleSymbolA.hasValue());
+    EXPECT_TRUE(doubleSymbolBBound.hasValue());
+}
+TEST_F(SymbolFixture, setValueAndGetValue)
+{
+    // In practice, setValue is tested for the most part in the test fixture's 
```

Comment:
This should be avoided. As much as possible we want tests to test functionality. It is better to keep fixtures to the minimum and then test functionality. As mentioned you can keep extending a fixture. A problem will arise if a bug in `setValue` happens then _all_ of your tests will fail since they rely on the same fixture. This will make debugging harder.